javascript/javascript2/week2/script.js

```diff
@@ -0,0 +1,193 @@
+(function () {
+  "use strict";
+  // ###### 1.1. Doubling of number
+
+  let numbers = [1, 2, 3, 4];
+  let newNumbers = [];
+
+  numbers.forEach((i) => {
+    if (i % 2 !== 0) {
+      newNumbers.push(i * 2);
+    }
+  });
+
+  // console.log("The doubled numbers are", newNumbers); // [2, 6]
+
+  // ###### 1.2. Working with movies
+
+  // STEP: 1 & 2
+  document
+    .querySelector("button")
+    .addEventListener("click", function getOption() {
+      // declaring variables for getting values for length of the title
+      const minTitleLength = document.querySelector("#min-title-length").value;
+      const maxTitleLength = document.querySelector("#max-title-length").value;
+      // new arrays for title lengths
+      function creatingTitlelength() {
+        if (minTitleLength !== "" && maxTitleLength !== "") {
+          const warning = document.createElement("h2");
+          warning.innerHTML = "Please only fill one option";
+          document.body.appendChild(warning);
+        }
+      }
+      creatingTitlelength();
+      const titleLength = movies.filter((i) => {
+        if (minTitleLength !== "" && maxTitleLength === "") {
+          return i.title.length < minTitleLength;
+        } else if (maxTitleLength !== "" && minTitleLength === "") {
+          return i.title.length > maxTitleLength;
+        }
+      });
+
+      //    console.log("MOVIES' BY TITLE", titleLength);
+
+      //STEP : 3
+      // variables for date filter
+
+      const fromDate = document.querySelector("#fromDate").value;
+      const toDate = document.querySelector("#toDate").value;
+
+      const dateFilter = movies.filter(
+        (i) => i.year > fromDate && i.year < toDate
+      );
+      //console.log(dateFilter);
+      // STEP: 4
+      // variables for rating filter
+      const ratingScore = document.querySelector("#rating-filter").value;
+      const ratingScroreNumFrom = document.querySelector(
+        "#rating-filter-score-from"
+      ).value;
+      const ratingScroreNumTo = document.querySelector(
+        "#rating-filter-score-to"
+      ).value;
+      const ratingScroreNum = dateFilter.filter(
+        (i) => i.rating >= ratingScroreNumFrom && i.rating <= ratingScroreNumTo
+      );
+
+      const badRating = dateFilter.filter((i) => i.rating < 4);
+      const averageRating = dateFilter.filter(
+        (i) => i.rating >= 4 && i.rating < 7
+      );
+      const goodRating = dateFilter.filter((i) => i.rating >= 7);
+
+      function ratingsFilter() {
+        if (ratingScroreNumFrom === "--" && ratingScroreNumTo === "--") {
+          const ratingDisplay = document.createElement("div");
+          document.body.append(ratingDisplay);
+          ratingDisplay.innerHTML = "<h2>Movies</h2>";
+          let ratingDisplayMovies = document.createElement("ol");
+          ratingDisplay.appendChild(ratingDisplayMovies);
```

Comment:
To avoid code duplication you could create a function taking the different arrays of movies (badRating, averageRating and goodRating). This function would be processing the different arrays and creating the li elements with the corresponding content
Example: 
 ```
 const displayMoviesByRating = (moviesArray) => {
         for (let i = 0; i < moviesArray.length; i++) {
               let ratingDisplayMoviesList = document.createElement("li");
               ratingDisplayMovies.append(ratingDisplayMoviesList);
               ratingDisplayMoviesList.innerHTML = moviesArray[i].title;
          }
}
```  
And then use this function inside the different if statements:
```
if (ratingScore === "Bad") {
        displayMoviesByRating(badRating);
} else if (ratingScore === "Average") {
        displayMoviesByRating(averageRating);
} else {
        displayMoviesByRating(goodRating);
}
```